src/realm/keys.hpp

```diff
@@ -266,11 +266,15 @@ struct ObjLink {
     }
     explicit operator bool() const
     {
-        return m_table_key.operator bool();
+        return bool(m_table_key) && bool(m_obj_key);
```

Comment:
Why was this change necessary?

Reply:
It was just found while figuring out why something else wasn't working. Constructing an ObjLink from a NULL ObjKey but a non-null TableKey should produce a NULL ObjLink.